pennylane/transforms/decompositions/two_qubit_unitary.py

```diff
@@ -611,9 +611,8 @@ def two_qubit_decomposition(U, wires):
     # If there is an active tape, queue the decomposition so that expand works
     current_tape = qml.tape.get_active_tape()
 
-    if current_tape:
-        with current_tape:
-            for op in decomp:
-                qml.apply(op)
+    if current_tape is not None:
+        for op in decomp:
+            qml.apply(op, context=current_tape)
```

Comment:
This is the only change?

🤔 🤔 🤔 🤔 🤔

Reply:
Hmm, the logic of the `qml.apply` function is quite different if `context` is passed!

```python
if hasattr(op, "queue"):
    # operator provides its own logic for queuing
    op.queue(context=context)
else:
    # append the operator directly to the relevant queuing context
    context.append(op)
```

My guess is that, in the old version, the `with current tape:` was causing the issue somehow?

Reply:
> This is the only change?

I know right? It's almost concerning :fearful: I was thinking in the old version maybe the contexts get weirdly nested somehow. Like we were already in a context, and then were doing, 'within the same context, do more stuff', and after a sufficient number of instances of that we hit the limit.